test/integration/support.js

```diff
@@ -2,14 +2,29 @@
 
 const Support = require('../support');
 
-const runningQueries = new Set();
+const runningQueries = new Map();
+const runningTransactions = new Map(); // map transaction option to queries.
 
 before(function() {
-  this.sequelize.addHook('beforeQuery', (options, query) => {
-    runningQueries.add(query);
+  this.sequelize.addHook('transactionCreated', t => { // tracking race condition, remove me if no longer present.
+    t.trace = new Error().stack;
   });
-  this.sequelize.addHook('afterQuery', (options, query) => {
+  this.sequelize.addHook('beforeQuery', (options, query, sql) => {
+    runningQueries.set(query, options);
+    if (options.transaction) {
+      const queryList = runningTransactions.get(options.transaction.id);
+      if (queryList) {
+        queryList.push(sql);
+      } else {
+        runningTransactions.set(options.transaction.id, [sql]);
+      }
+    }
+  });
+  this.sequelize.addHook('afterQuery', (options, query, sql) => {
     runningQueries.delete(query);
+    if (options.transaction && sql.includes('COMMIT')) {
```

Comment:
should this better be:?

```suggestion
    if (options.transaction && ['COMMIT', 'ROLLBACK'].includes(sql)) {
```